server/src/main/java/org/elasticsearch/health/GetHealthAction.java

```diff
@@ -44,15 +44,21 @@ private GetHealthAction() {
         private final ClusterName clusterName;
         private final HealthStatus status;
         private final List<HealthComponentResult> components;
+        private final boolean showTopLevelStatus;
 
         public Response(StreamInput in) {
             throw new AssertionError("GetHealthAction should not be sent over the wire.");
         }
 
-        public Response(final ClusterName clusterName, final List<HealthComponentResult> components) {
-            this.clusterName = clusterName;
+        public Response(final ClusterName clusterName, final List<HealthComponentResult> components, boolean showTopLevelStatus) {
+            this.showTopLevelStatus = showTopLevelStatus;
```

Comment:
can we mark the `status` as `@Nullable` and instead of using the `showTopLevelStatus` flag check the status for `null` ?
This will remove the need to have the `showTopLevelStatus` as a field but only as constructor parameter - which, would it be better named as `isDrillDown` or something similar?

Reply:
Done.